spec/unit/outputs/elasticsearch_spec.rb

```diff
@@ -73,6 +93,27 @@
       end
     end
 
+    context "on a reachable ES instance" do
+      let(:events) { [ ::LogStash::Event.new("foo" => "bar1"), ::LogStash::Event.new("foo" => "bar2") ] }
+
+      let(:shutdown_value) { true }
+
+      let(:logger) { double("logger") }
+      before(:each) do
+        allow(subject).to receive(:logger).and_return(logger)
+        allow(logger).to receive(:info)
+
+        allow(subject).to receive(:pipeline_shutdown_requested?) do
+          shutdown_value
+        end
+        allow(subject).to receive(:retrying_submit)
+      end
+
+      it "the #multi_receive doesn't abort when waiting for a connection on alive ES and a shutdown is requested" do
+        expect(logger).to_not have_received(:info).with(/Aborting the batch due to shutdown request while waiting for connections to become live/i)
+      end
+    end
```

Comment:
```suggestion
      context "when waiting for a connection on alive ES and a shutdown is requested" do
        describe "#multi_receive" do
          it "doesn't abort" do
            expect(logger).to_not have_received(:info).with(/Aborting the batch due to shutdown request while waiting for connections to become live/i)
          end 
        end
      end
    end
```

Reply:
Hi @jsvd with commit ece4c94fb5a5121faf7dd0a8c319c815ff23b0ff I've updated the test, adding the invocation to `multi_receive` and commented the fix to prove that the test fails.
With commit 7c4e477ad34398cf81a7a5793cb99e1472d1d2af I've reenabled the fix to check the tests becomes green, however the CI remains stuck in some jobs as described in [comment](https://github.com/logstash-plugins/logstash-output-elasticsearch/pull/1132#issuecomment-1568367270).
The same error appears also on `main` so should be related to latest changes in `8.8.1`on the ES side, I think.
The PR is ready for another review.